src/http_fork_source.rs

```diff
@@ -0,0 +1,53 @@
+use zksync_web3_decl::{
+    jsonrpsee::http_client::{HttpClient, HttpClientBuilder},
+    namespaces::{EthNamespaceClient, ZksNamespaceClient},
+};
+
+use crate::fork::{block_on, ForkSource};
+
+#[derive(Debug)]
+/// Fork source that gets the data via HTTP requests.
+pub struct HttpForkSource {
+    pub fork_url: String,
+}
+
+impl HttpForkSource {
+    pub fn create_client(&self) -> HttpClient {
+        HttpClientBuilder::default()
+            .build(self.fork_url.clone())
+            .expect("Unable to create a client for fork")
+    }
+}
+
+impl ForkSource for HttpForkSource {
+    fn get_storage_at(
+        &self,
+        address: zksync_basic_types::Address,
+        idx: zksync_basic_types::U256,
+        block: Option<zksync_types::api::BlockIdVariant>,
+    ) -> zksync_basic_types::H256 {
+        let client = self.create_client();
+        block_on(async move { client.get_storage_at(address, idx, block).await }).unwrap()
```

Comment:
We should handle errors better and avoid using `.unwrap()`.

Should we update the responses for `get_storage_at`, `get_bytecode_by_hash`, `get_transaction_by_hash` and `get_raw_block_transactions` to have `Result<...>` values instead?

Reply:
done. For now I'm still doing unwraps on the higher level (as it requires changing other signatures - that can be done in separate PR).